flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraph.java

```diff
@@ -148,6 +162,33 @@ public Logger getLogger() {
         return logger;
     }
 
+    protected Throwable extractError(TaskExecutionStateTransition taskExecutionStateTransition) {
+        Throwable cause = taskExecutionStateTransition.getError(userCodeClassLoader);
+        if (cause == null) {
+            cause = new FlinkException("Unknown failure cause. Probably related to FLINK-21376.");
+        }
+        return cause;
+    }
+
+    protected Optional<ExecutionVertexID> extractExecutionVertexID(
+            TaskExecutionStateTransition taskExecutionStateTransition) {
+        return executionGraph.getExecutionVertexId(taskExecutionStateTransition.getID());
```

Comment:
If you want to go this way, I'd rather do something along the lines of `Optional<Execution> ExecutionGraph#getExecution(...)`, because that's something we'd be able to reuse on other places immediately;

But feel free to ignore this, there's nothing wrong with this approach; I mostly feel uneasy about adding new methods to already overloaded interfaces, but in this case it doesn't add complexity